### PR TITLE
 Replace deprecated `props.url` in readme

### LINKED ISSUE #4716: Outdated references to props.url in README
@timneutkens The repo's `README` states: `Note: in order to programmatically change the route without triggering navigation and component-fetching, use props.url.push and props.url.replace within a component`

Since `url` as a prop has been deprecated in Next v6, the file should be updated to indicate the correct way of performing a programatic url update with `router`.

It would be nice to also have an example that shows how to use the `as` parameter as well. In @tim's own words:
>  Basically `push(href, as)`
>   `href` => page inside `pages` directory
>    `as` => what’s displayed in the browser

----

https://github.com/zeit/next.js#routing is also including this bit `Deprecated, use withRouter instead - Each top-level component receives a url property with the following API`, which I believe is confusing.

 It would be much better phrased if mentioned straight away the correct API with `withRouter` (or alternatively even passing down `router` as a prop from the page-level component) and mention that in previous versions `url` was available as a prop, but that's not the case anymore.

-----

There is also a reference to using `url.pushTo`.

----

And `url` is also mentioned as a prop in the [shallow routing](https://github.com/zeit/next.js#shallow-routing) section: `You'll receive the updated pathname and the query via the url prop of the same page that's loaded, without losing state.`

### PR DESCRIPTION
**Pull request by lucleray**
Following https://github.com/zeit/next.js/pull/4950

Fix https://github.com/zeit/next.js/issues/4716

I'm not sure to understand this part, though :
> Note: in order to programmatically change the route without triggering navigation and component-fetching, use `props.url.push` and `props.url.replace` within a component

Is there a difference between :
```jsx
export default () => <a onClick={() => Router.push('/about')}>About</a>
```

and
 
```jsx
export default withRouter(
  () => <a onClick={() => props.router.push('/about')}>About</a>
)
```
?

### CODE REVIEW

**Comment by timneutkens** [comment]
> Note: in order to programmatically change the route without triggering navigation and component-fetching, use props.url.push and props.url.replace within a component

This has been deprecated for ages (Since 2.0 was released or so), in general I would recommend using `withRouter` for everything as that follows React's component model vs `import Router from 'next/router'` being stateful/global.

**Comment by lucleray** [comment]
@timneutkens 

Ok, I'll remove the note and I'll update the examples in the Readme accordingly 👍

File: readme.md [hunk 1]
```diff
@@ -708,9 +704,9 @@ Most prefetching needs are addressed by `<Link />`, but we also expose an impera
 ```jsx
 import Router from 'next/router'
```
Comment (timneutkens) [inline readme.md]: Lets change this to use `withRouter` instead 👍

**Comment by lucleray** [comment]
I still feel like the Router API is a bit hidden in the part about "Intercepting `popstate`".

I think it would be good to move it to "Imperatively" just above, or even to move it to "withRouter", since it's the recommended way of using the router 😄

**Review by timneutkens — APPROVED** [review]

**Pull request merged**